File: src/entity/player/Player.java
package entity.player;

import com.raylib.Jaylib;
import com.raylib.Raylib;
import com.raylib.Raylib.Texture;
import com.raylib.Raylib.Vector2;

import asset.Assets;
import main.Utils;
import vector.Vector2f;

public class Player {

	// Instance stuff
	public Vector2 position = new Vector2(0);
	public Vector2f movement = new Vector2f(0f, 0f);
	public int width = 256;
	public int height = 256;
	public int controllerIndex;
	public final float speed = 5f;
	public Texture texture = null;

	public Player(int controller) {

		
		controllerIndex = controller;
		this.texture = Assets.player;
		this.texture.width(this.width);
		this.texture.height(this.height);

		System.out.println("Adding a new player with controller index of " + controllerIndex);
	}



	public final void getInput() {

		// Check for if the controller is avalible
		if (!Raylib.IsGamepadAvailable(controllerIndex)) return;

		// Get input
		this.movement.set(

			Raylib.GetGamepadAxisMovement(this.controllerIndex, 0) * this.speed,
			Raylib.GetGamepadAxisMovement(this.controllerIndex, 1) * this.speed

		);

		this.position.x(this.position.x() + this.movement.x);
		this.position.y(this.position.y() + this.movement.y);

	}



	public final void update() {

		getInput();

		// System.out.println(this.position.x() + ", " + this.position.y());
		System.out.println(Raylib.GetGamepadAxisMovement(this.controllerIndex, 0) + ", " + Raylib.GetGamepadAxisMovement(this.controllerIndex, 1));

	}



	public final void render() {

		Raylib.DrawTextureV(this.texture, this.position, Jaylib.WHITE);

		// Utils.DrawTextWithController("Press <down_x> or <down_ps> to join", 10, 10, 35);
	}

}

File: src/entity/player/Player.cs
using System.Numerics;
using Raylib_cs;

class Player
{
	public int ControllerIndex { get; private set; }
	public Vector2 Position { get; set; }
	public Texture2D Texture { get; private set; }

	private readonly float speed = 200f;

	public Player(int controllerIndex)
	{
		// Assign a heap of stuff
		// TODO: Use asset manager
		ControllerIndex = controllerIndex;
		
		// Set the player texture depending on the
		// controller index
		// TODO: Use lookup table
		switch (controllerIndex)
		{
			case 0:
				Texture = Assets.Player1;
				break;

			case 1:
				Texture = Assets.Player2;
				break;

			case 2:
				Texture = Assets.Player3;
				break;

			case 3:
				Texture = Assets.Player4;
				break;
		}

		Console.WriteLine($"Player with controller index {ControllerIndex} registered!!");
	}

	public void Update()
	{
		Movement();
	}

	public void Render()
	{
		Raylib.DrawTextureV(Texture, Position, Color.White);
	}





	private void Movement()
	{
		// Check for if the controller is available
		if (!Raylib.IsGamepadAvailable(ControllerIndex)) return;

		// Get the input input from the analog stick
		float xAxis = Raylib.GetGamepadAxisMovement(ControllerIndex, GamepadAxis.LeftX);
		float yAxis = Raylib.GetGamepadAxisMovement(ControllerIndex, GamepadAxis.LeftY);

		// Apply speed and delta time
		Vector2 movement = new Vector2(xAxis, yAxis) * speed * Raylib.GetFrameTime();

		// Update the players position
		Position += movement;
	}
}